_source/_posts/2022-09-26-java-jetty-oauth.md

```diff
@@ -0,0 +1,1017 @@
+
+Update the `WebController` to the following.
+
+`src/main/java/com/demo/WebController.java`
+
+```java
+package com.demo;
+
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Controller
+@RequestMapping(path = "/hikes")
+public class WebController {
+
+   private List<String> hikes = new ArrayList<>(Arrays.asList(
+            "Wonderland Trail", "South Maroon Peak", "Tour du Mont Blanc",
+            "Teton Crest Trail", "Everest Base Camp via Cho La Pass", "Kesugi Ridge"
+    ));
+
+    @GetMapping("")
+    @ResponseBody
+    public String indexGet() {
+        return String.join("\n", this.hikes);
+    }
+
+    @PostMapping("")
+    @ResponseBody
+    @PreAuthorize("isAuthenticated")  // <- ***ADDED***
+    public String indexPost(@RequestParam String hike, HttpServletResponse response) {
+        if (hike == null) {
+            response.setStatus(400);
+            return "Param 'hike' cannot be null.";
+        }
+        else if (this.hikes.contains(hike)) {
+            response.setStatus(400);
+            return "The hike '"+hike+"' already exists.";
+        }
+        else {
+            this.hikes.add(hike);
+            return String.join("\n", this.hikes);
+        }
+    }
+
+    @DeleteMapping("")
+    @ResponseBody
+    @PreAuthorize("isAuthenticated")  // <- ***ADDED***
+    public String indexDelete(@RequestParam String hike, HttpServletResponse response) {
+        if (hike == null) {
+            response.setStatus(400);
+            return "Param 'hike' cannot be null.";
+        }
+        
+        
+        else {
+            this.hikes.remove(hike);
+            return String.join("\n", this.hikes);
+        }
+    }
+
+}
+```
+
+Try to POST a new hike.
+
+```bash
+http -f POST :8080 hike="Pear Lake"
+```
+
+```bash
+HTTP/1.1 403 Forbidden
+...
+
+{
+    "error": "Forbidden",
+    "path": "/",
+    "status": 403,
+    "timestamp": "2022-09-06T14:09:02.533+00:00"
+}
+```
+
+## Generate a JWT using the OIDC debugger
+
+To access the protected endpoints, you need to generate an access token JWT. {% include setup/oidcdebugger.md %}
+
+{% img blog/java-jetty/oidc-debugger.png alt:"OIDC Debugger Configuration" width:"650" %}{: .center-image }
+
+Scroll down and click **Send Request**.
+
+This will respond with an authorization code, which you need to send back to the authorization URI to retrieve the code.
```

Comment:
This seems harder than it needs to be. Maybe it's because you're combining OIDC Debugger and Spring Boot in the same app? Here's what I've done in a previous post: https://developer.okta.com/blog/2022/05/25/java-rest-api-spring-boot#generate-an-oauth-2-0-access-token